tests/unit/test_reader_v1.py

```diff
@@ -198,22 +225,21 @@ def test_rows_w_timeout(class_under_test, mock_gapic_client):
     bq_blocks_2 = [[{"int_col": 567}, {"int_col": 789}], [{"int_col": 890}]]
     avro_blocks_2 = _bq_to_avro_blocks(bq_blocks_2, avro_schema)
 
-    mock_gapic_client.read_rows.return_value = avro_blocks_2
+    mock_gapic_client.read_rows.side_effect = (
+        avro_blocks_1,
+        avro_blocks_2,
+    )
 
     reader = class_under_test(
-        avro_blocks_1,
-        mock_gapic_client,
-        "teststream",
-        0,
-        {"metadata": {"test-key": "test-value"}},
+        mock_gapic_client, "teststream", 0, {"metadata": {"test-key": "test-value"}},
     )
 
     with pytest.raises(google.api_core.exceptions.DeadlineExceeded):
         list(reader.rows())
 
     # Don't reconnect on DeadlineException. This allows user-specified timeouts
     # to be respected.
-    mock_gapic_client.read_rows.assert_not_called()
+    mock_gapic_client.read_rows.assert_called_once()
```

Comment:
Comment is out-of-date now. I believe this is the breaking change side-effect I mentioned earlier.

I'd prefer we don't break this behavior and find a way to implement the reconnect logic in `BigQueryReadClient.read_rows` and `ReadRowsStream`. Perhaps a helper function could be created to keep it DRY?

Reply:
Removed comment. As mentioned above, behavior is the same as before.